### PR TITLE
USPS International Rates

### PR DESCRIPTION
**Pull request by barberj**
USPS International

Add support for requesting rate estimates for USPS International packages

### CODE REVIEW

**Comment by pelargir** [comment]
This is looking good.

File: lib/friendly_shipping/services/usps_international/parse_package_rate.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          flat_rate_envelope: 'Flat Rate Envelope',
+          legal_flat_rate_envelope: 'Legal Flat Rate Envelope',
+          padded_flat_rate_envelope: 'Padded Flat Rate Envelope',
+          gift_card_flat_rate_envelope: 'Gift Card Flat Rate Envelope',
+          window_flat_rate_envelope: 'Window Flat Rate Envelope',
+          small_flat_rate_envelope: 'Small Flat Rate Envelope',
+          large_envelope: 'Large Envelope',
+          parcel: 'Parcel',
+          postcards: 'Postcards'
+        }.map { |k, v| "(?<#{k}>#{v})" }.join("|").freeze
```
Comment (barberj) [inline lib/friendly_shipping/services/usps_international/parse_package_rate.rb:36]: @pelargir this is more or less me imitating what i saw in Usps Domestic. Not really sure what the value is with the code here. I did pair down to what i believe is valid for international.

File: lib/friendly_shipping/services/usps_international/parse_rate_response.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            parsing_result.fmap do |xml|
+              # Get all the possible rates for each package
+              rates_by_package = rates_from_response_node(xml, shipment, options)
+
+              rates = SHIPPING_METHODS.map do |shipping_method|
+                # For every package ...
+                matching_rates = rates_by_package.map do |package, package_rates|
+                  # ... choose the rate that matches the shipping method for this package
+                  package_rates.select { |r| r.shipping_method == shipping_method }.first
+                end.compact # Some shipping rates are not available for every shipping method.
```
Comment (barberj) [inline lib/friendly_shipping/services/usps_international/parse_rate_response.rb:36]: returning them all. I'm expecting caller to sort out. We could find best by the shipping method for the package? My experience there is only ever one rate per package per shipping method...

**Review by mamhoff — APPROVED** [review]

File: lib/friendly_shipping/services/usps_international/parse_rate_response.rb (hunk 2 above)
Comment (pelargir) [reply lib/friendly_shipping/services/usps_international/parse_rate_response.rb:36]: Returning them all is fine.

**Review by pelargir — COMMENTED** [review]
Great work on this. 💯 

I have one suggestion regarding Pry. I think it would also be helpful to amend the very last commit message ("Refactor") to include a brief description of what/why.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@
   add_filter '/spec/'
 end
 
+require 'pry'
```
[the remark this answers is not in the document]
Comment (barberj) [reply spec/spec_helper.rb:8]: It is in the gemspec file.... without requiring it in spec_helper i get an exception

```
    has the correct error (FAILED - 1)

Failures:

  1) FriendlyShipping::Services::Ups::ParseXMLResponse with invalid XML in response body has the correct error
     Failure/Error: binding.pry

     NoMethodError:
       undefined method `pry' for #<Binding:0x00000001096ebbd0>

       binding.pry
              ^^^^
       Did you mean?  try
```

[the remark this answers is not in the document]
Comment (pelargir) [reply spec/spec_helper.rb:8]: Doh, I missed it completely. 👍

**Pull request merged**